src/control/lib/hardware/topology.go

```diff
@@ -47,6 +47,27 @@ func (t *Topology) AllDevices() map[string]*PCIDevice {
 	return devsByName
 }
 
+// NumNUMANodes gets the number of NUMA nodes in the system topology.
+func (t *Topology) NumNUMANodes() int {
+	if t == nil {
+		return 0
+	}
+	return len(t.NUMANodes)
+}
+
+// NumCoresPerNUMA gets the number of cores per NUMA node.
+func (t *Topology) NumCoresPerNUMA() int {
```

Comment:
I'm not sure about telling the caller what to do in this situation. In practice I this is used as part of a gRPC message. This may go away entirely, as the assumption that all NUMA nodes are the same may not be valid anyway. I'd prefer to iterate on this in the future and figure out how we should be presenting this information.